### PR TITLE
[BEAM-9703] Fix user distribution metric test by not validating namespace.

### PR DESCRIPTION
**Pull request by HuangLED**
Previous validation was too strong.  
1) Namespace differs by testing environments (runners). Use contain substring instead in this PR. 
2) min value is too restrictive for streaming case (where the metrics may get reported as best effort)

Alternative way would be to completely disable this test for streaming (and create a dedicated one). 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by HuangLED** [comment]
R: @tvalentyn , @lukecwik

**Comment by HuangLED** [comment]
R: @tvalentyn, @lukecwik

**Comment by tvalentyn** [comment]
Run tests

**Comment by lukecwik** [comment]
Please run yapf to fix the python formatting.

**Comment by HuangLED** [comment]
Did run gradlew lint before commit. Looks like yaqf does things slightly different from gradlew-lint. 

Now applied "yapf --in-place apache_beam/metrics/metric_test.py".  PTAL.

**Comment by lukecwik** [comment]
retest this please

File: sdks/python/apache_beam/metrics/metric_test.py [hunk 1]
```diff
@@ -172,10 +172,13 @@ def process(self, element):
     # Verify user distribution counter.
     metric_results = res.metrics().query()
     matcher = MetricResultMatcher(
-        namespace='apache_beam.metrics.metric_test.SomeDoFn',
+        step='ApplyPardo',
+        namespace=hc.contains_string('SomeDoFn'),
         name='element_dist',
         committed=DistributionMatcher(
-            sum_value=10, count_value=4, min_value=1, max_value=4))
+            sum_value=10, count_value=4,
```
Comment (lukecwik) [inline sdks/python/apache_beam/metrics/metric_test.py]: why would min value be >= 0 but the other values be exact?

Comment (HuangLED) [reply sdks/python/apache_beam/metrics/metric_test.py]: b/c after repeating it multiple times, streaming side gives consistent outcome on these exact numbers. 

After another thought though, given how much uncertainties there are on streaming side, it is really not necessary to put exact number here. And if we do really want to check exact number, there is alternative way to add that separately. 

Updated by making all other threes to be 'greater_than(0)'.  PTAL.

File: sdks/python/apache_beam/metrics/metric_test.py [hunk 2]
```diff
@@ -172,10 +172,14 @@ def process(self, element):
...
     matcher = MetricResultMatcher(
-        namespace='apache_beam.metrics.metric_test.SomeDoFn',
+        step='ApplyPardo',
+        namespace=hc.contains_string('SomeDoFn'),
         name='element_dist',
         committed=DistributionMatcher(
-            sum_value=10, count_value=4, min_value=1, max_value=4))
+            sum_value=hc.greater_than(0),
+            count_value=hc.greater_than(0),
+            min_value=hc.greater_than_or_equal_to(0),
```
Comment (lukecwik) [inline sdks/python/apache_beam/metrics/metric_test.py:181]: Why is min different then the others?

I would expect that the same reason this needs to be >= 0 would be the same reason as the others.

Comment (HuangLED) [reply sdks/python/apache_beam/metrics/metric_test.py:181]: Fair enough. 

My justification was that as long as there is one single update picked up, then other values should be something non-zero. (e.g. count would be at least one).

Maybe I still underestimated how much uncertainties there is. 

Updated.  Now they are all >=0.

**Comment by lukecwik** [comment]
retest this please

**Comment by lukecwik** [comment]
retest this please

**Comment by tvalentyn** [comment]
@lukecwik  - does latest revision address your comments?

**Pull request merged**